File: OOP/employee-salary-management-app/src/com/techlab/employee/Employee.java
package com.techlab.employee;

public abstract class Employee {
	private String name;
	private long number;
	private double basicSalary;
	private double netSalary;

	public Employee(String name, long number, double basicSalary) {
		this.name = name;
		this.number = number;
		this.basicSalary = basicSalary;
	}

	public abstract void calculateNetSalary();

	public double getNetSalary() {
		return netSalary;
	}

	public void setNetSalary(double netSalary) {
		this.netSalary = netSalary;
	}

	public String getName() {
		return name;
	}

	public long getNumber() {
		return number;
	}

	public double getBasicSalary() {
		return basicSalary;
	}
}

File: nodejs/typescript/Employee.ts
export class Employee {
    constructor(private id: number, private name: string, private address: object) {

    }
    public get Id() {
        return this.id;
    }
    public set Id(id: number) {
        this.id = id;
    }
}

export class Address {
    constructor(private city: string, private state: string) {
    }
}